### PR TITLE
chore: update dependancies

### PR DESCRIPTION
**Pull request by pete-watters**
> Try out Leather build 6721866 — [Extension build](https://github.com/leather-wallet/extension/actions/runs/9561340803), [Test report](https://leather-wallet.github.io/playwright-reports/chore-update-dependancies), [Storybook](https://chore-update-dependancies--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=chore-update-dependancies)

This PR updates dependancies from:
- https://github.com/leather-wallet/extension/pull/5543
- https://github.com/leather-wallet/extension/pull/5542
- https://github.com/leather-wallet/extension/pull/5541/
- https://github.com/leather-wallet/extension/pull/5540/
- https://github.com/leather-wallet/extension/pull/5539/

## Summary by CodeRabbit

- **Chores**
	- Updated GitHub Actions workflows to use the latest version (`v6`) of `dawidd6/action-download-artifact`.
	- Updated various dependencies in `package.json`:
		- `@leather-wallet/bitcoin` to version "0.7.3"
		- `@leather-wallet/constants` to version "0.6.4"
		- `@leather-wallet/query` to version "0.8.8"
		- `@leather-wallet/utils` to version "0.8.2"

### CODE REVIEW

**Comment by pete-watters** [comment]
Audit is failing here for https://github.com/advisories/GHSA-3h5v-q93c-6h6q

File: package.json [hunk 1]
```diff
@@ -366,7 +366,8 @@
     "socket.io-parser": "4.2.4",
     "wrap-ansi": "7.0.0",
     "webpack-dev-middleware": "5.3.4",
-    "eslint": "8.56.0"
+    "eslint": "8.56.0",
+    "ws": "8.17.1"
```
Comment (kyranjamie) [inline package.json:370]: Should we be adding this as a dep, or setting a resolution? Afaik, if we don't use directly, but want a later version, it should be a resolution

Comment (pete-watters) [reply package.json:370]: I couldn't find us using it directly, so I added it as a resolution.

Comment (kyranjamie) [reply package.json:370]: My bad, didn't see it was in resolutions

**Review by kyranjamie — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (pete-watters) [reply package.json:370]: No problem. I wasn't certain when you said it so I double checked which is worth it 👍

**Pull request merged**